### PR TITLE
SDA-7306 Update NodePool model with status attributes

### PR DESCRIPTION
**Pull request by lucasponce**

### CODE REVIEW

File: model/clusters_mgmt/v1/node_pool_type.model [hunk 1]
```diff
@@ -38,4 +41,7 @@ class NodePool {
 
     // The subnet upon which the nodes are created.
     Subnet String
+
+    // Adds additional information about the NodePool status when Replicas != CurrentReplicas.
+    Message String
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/node_pool_type.model:46]: Perhaps this should be a struct?
Also it sounds very generic. Can you provide an example of what this msg will be?

Comment (lucasponce) [reply model/clusters_mgmt/v1/node_pool_type.model:46]: The message is open.
NodePools don't have a lifecycle; the status is the CurrentReplicas; this message is a custom message that would explain the situation.

It's generic. Then a string is the best option for this case.

**Review by tbrisker — COMMENTED** [review]
I don't see this in the backend yet, normally we update the api model after the backend is merged

File: model/clusters_mgmt/v1/node_pool_type.model (hunk 1 above)
[the remark this answers is not in the document]
Comment (lucasponce) [reply model/clusters_mgmt/v1/node_pool_type.model:46]: I don't think so.
There is no lifecycle.
So, the strategy I'm following in this iteration is to explain why the CurrentReplicas don't match the desired one.

If a former "status" is defined in hypershift then we can reevaluate, but I'd prefer not to introduce an artificial lifecycle.

**Comment by lucasponce** [comment]
> I don't see this in the backend yet, normally we update the api model after the backend is merged

Ah good point, thanks for the pointer. 
I'll hold on this PR until the backend is good.

File: model/clusters_mgmt/v1/node_pool_type.model (hunk 1 above)
Comment (lucasponce) [reply model/clusters_mgmt/v1/node_pool_type.model:46]: But +1.
It makes sense if status grows.
Thanks

[the remark this answers is not in the document]
Comment (lucasponce) [reply model/clusters_mgmt/v1/node_pool_type.model:46]: I was biased in my first thought as I was mimicking other structures.
But yes, let's wrap the message inside a status struct.

**Comment by lucasponce** [comment]
cc @vkareh 

This are the changes needed in the model to update the ROSA CLI with the new NodePool status.

File: CHANGES.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 This document describes the relevant changes between releases of the API model.
 
+## 0.0.250 Jan 12 2023
+- Add `NodePoolStatus` type to the `NodePool` resource model.
```
Comment (vkareh) [inline CHANGES.md:6]: We usually add these during the release tagging, after the actual changes have been merged, so you should probably remove this change.

File: model/clusters_mgmt/v1/node_pool_status_type.model [hunk 3]
```diff
@@ -0,0 +1,24 @@
+/*
+Copyright (c) 2022 Red Hat, Inc.
```
Comment (vkareh) [inline model/clusters_mgmt/v1/node_pool_status_type.model:2]: 2023

File: model/clusters_mgmt/v1/node_pool_type.model [hunk 4]
```diff
@@ -23,6 +23,9 @@ class NodePool {
     // Replicas and autoscaling cannot be used together.
     Replicas Integer
 
+    // The number of Machines (and Nodes) currently created.
+    CurrentReplicas Integer
```
Comment (vkareh) [inline model/clusters_mgmt/v1/node_pool_type.model:27]: This is outdated, remove.

**Comment by vkareh** [comment]
LGTM

**Comment by lucasponce** [comment]
Not yet ready, it requires an update in the Cluster Status.

**Comment by ciaranRoche** [comment]
lgtm

**Pull request merged**